table/tables/tables.go

```diff
@@ -1008,8 +1030,17 @@ func (t *TableCommon) RemoveRecord(ctx sessionctx.Context, h kv.Handle, r []type
 		return err
 	}
 
+	txn, err := ctx.Txn(true)
+	if err != nil {
+		return err
+	}
 	if m := t.Meta(); m.TempTableType == model.TempTableGlobal {
-		addTemporaryTable(ctx, m)
+		if tmpTable := addTemporaryTable(ctx, m); tmpTable != nil {
```

Comment:
Can we use a function to do these duplicate things?

Reply:
The defer function have to be there ... because we calculate the txn delta size before and after the execution.
If it's put inside another function, `defer` does not work anymore.
```
if tmpTable.GetSize() > ctx.GetSessionVars().TempTableMaxRAM {
        return errors.New("temporary table size exceed @@global.temptable_max_ram")
}
```

can be a function but it will not simplify the code, because of the notorious Go error handling:

```
if err := checkTmpTableSize(); err != nil {
        return err
}
```